client/lib/utils/test/error.js

```diff
@@ -0,0 +1,102 @@
+import { expect } from 'chai';
+import {
+	isFieldError,
+} from '../error';
+
+describe( 'Error utils', () => {
+	it( '#isFieldError(): required, no pattern, empty value', () => {
+		const required = true;
+		const schema = {
+			pattern: '',
+		};
+		const value = '';
+
+		const isError = isFieldError( required, schema, value );
+
+		expect( isError ).to.be.true;
+	} );
+
+	it( '#isFieldError(): required, no pattern, not empty value', () => {
+		const required = true;
+		const schema = {
+			pattern: '',
+		};
+		const value = '1234';
+
+		const isError = isFieldError( required, schema, value );
+
+		expect( isError ).to.be.false;
+	} );
+
+	it( '#isFieldError(): required, pattern, empty value', () => {
+		const required = true;
+		const schema = {
+			pattern: /^1234/,
+		};
+		const value = '';
+
+		const isError = isFieldError( required, schema, value );
+
+		expect( isError ).to.be.true;
+	} );
+
+	it( '#isFieldError(): not required, no pattern, empty value', () => {
+		const required = false;
+		const schema = {
+			pattern: '',
+		};
+		const value = '';
+
+		const isError = isFieldError( required, schema, value );
+
+		expect( isError ).to.be.false;
+	} );
+
+	it( '#isFieldError(): not required, pattern, empty value', () => {
+		const required = false;
+		const schema = {
+			pattern: /^1234/,
+		};
+		const value = '';
+
+		const isError = isFieldError( required, schema, value );
+
+		expect( isError ).to.be.true;
+	} );
+
+	it( '#isFieldError(): not required, pattern, expected value', () => {
+		const required = false;
+		const schema = {
+			pattern: /^1234/,
+		};
+		const value = '1234';
+
+		const isError = isFieldError( required, schema, value );
+
+		expect( isError ).to.be.false;
+	} );
+
+	it( '#isFieldError(): not required, pattern, not expected value', () => {
+		const required = false;
+		const schema = {
+			pattern: /^1234/,
+		};
+		const value = '1235';
+
+		const isError = isFieldError( required, schema, value );
+
+		expect( isError ).to.be.true;
+	} );
+
+	it( '#isFieldError(): required, pattern, not expected value', () => {
+		const required = true;
+		const schema = {
+			pattern: /^1234/,
+		};
+		const value = '1235';
+
+		const isError = isFieldError( required, schema, value );
+
+		expect( isError ).to.be.true;
+	} );
+} );

```

Comment:
If one of these tests fail, its going to take a developer a little bit of time to figure out which of them to look at, since the names of the tests are so similar and there are so many of them.  May I suggest including a unique identifier in the description, e.g. "FE_TEST_1", "FE_TEST_2", etc